### PR TITLE
bugfix(integrations): Salesforce added network response handling 

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

Added response handling for Salesforce network requests. Correct parsing of retryable, abortable network response and returning same back to server

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/salesforce/transform.js [hunk 1]
```diff
@@ -39,20 +41,22 @@ async function getSFDCHeader(destination) {
   )}${encodeURIComponent(destination.Config.initialAccessToken)}&client_id=${
     destination.Config.consumerKey
   }&client_secret=${destination.Config.consumerSecret}&grant_type=password`;
-  let response;
-  try {
-    response = await axios.post(authUrl, {});
-  } catch (error) {
-    logger.error(error);
+  const salesforceAuthResponse = await httpPOST(authUrl, {});
```
Comment (saikumarrs) [inline v0/destinations/salesforce/transform.js:44]: Do we need "salesforce" in the variable names? They are making them longer. If it's unavoidable, consider using at least the shortform "sf"?

Comment (utsabc) [reply v0/destinations/salesforce/transform.js:44]: Make sense

**Review by ItsSudip — APPROVED** [review]

File: v0/destinations/salesforce/transform.js [hunk 2]
```diff
@@ -39,20 +41,20 @@ async function getSFDCHeader(destination) {
...
+  const sfAuthResponse = await httpPOST(authUrl, {});
+  const processedsfAuthResponse = processAxiosResponse(sfAuthResponse);
+  if (processedsfAuthResponse.status !== 200) {
     throw new CustomError(
-      `SALESFORCE AUTH FAILED: ${error.message}`,
-      error.status || 400
+      `SALESFORCE AUTH FAILED: ${JSON.stringify(
+        processedsfAuthResponse.response
+      )}`,
+      processedsfAuthResponse.status
```
Comment (arnab-p) [inline v0/destinations/salesforce/transform.js:51]: Should it be?
```suggestion
      processedsfAuthResponse.status || 400
```

Comment (utsabc) [reply v0/destinations/salesforce/transform.js:51]: `processAxiosResponse` response handles this there wont be a case where status would be undefined

**Review by arnab-p — APPROVED** [review]

File: v0/destinations/salesforce/transform.js (hunk 2 above)
Comment (arnab-p) [reply v0/destinations/salesforce/transform.js:51]: Cool

**Pull request merged**